src/Xamarin.Android.Build.Tasks/Tasks/CompileNativeAssembly.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using ThreadingTasks = System.Threading.Tasks;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	public class CompileNativeAssembly : AsyncTask
+	{
+		sealed class Config
+		{
+			public string AssemblerPath;
+			public string AssemblerOptions;
+			public string InputSource;
+		}
+
+		[Required]
+		public ITaskItem[] Sources { get; set; }
+
+		[Required]
+		public bool DebugBuild { get; set; }
+
+		[Required]
+		public string WorkingDirectory { get; set; }
+
+		public override bool Execute ()
+		{
+			try {
+				return DoExecute ();
+			} catch (Exception e) {
+				Log.LogCodedError ("XA3001", $"{e}");
```

Comment:
This should be a more "context free" error message.  We certainly *don't* want the stack trace dumped to the "error" output, as this is user hostile (wtf are they supposed to do with it?), though that said, what are they supposed to do when this task fails, *period*? :-/

Regardless, I'd suggest:

```suggestion
				Log.LogCodedError ("XA3001", $"Unable to generate `libxamarin-app.so`: {e.Message}");
```